lib/puppet/defaults.rb

```diff
@@ -1395,7 +1395,24 @@ def self.default_diffargs
     },
     :server => {
       :default => "puppet",
-      :desc => "The puppet master server to which the puppet agent should connect."
+      :desc => "The puppet master server to which the puppet agent should connect.",
+      :hook => proc { |value|
+          if Puppet.settings.set_by_config?(:server_list)
+            Puppet.deprecation_warning('Server_list has already been set. Server setting will not be used.')
+          end        
+      }
+    },
+    :server_list => {
+      :default => ["puppet"],
+      :type => :array,
+      :desc => "The list of puppet master servers to which the puppet agent should connect,
+        in the order that they will be tried.",
+      :hook => proc { |value| 
+            if Puppet.settings.set_by_config?(:server)  
```

Comment:
:+1: sweet.

Good to know our accurate naming conventions stand within this part of the codebase ;)